frontend/src/components/dialogs/AssetCompareDialog.vue

```diff
@@ -0,0 +1,158 @@
+                    <ff-dropdown-option
+                        v-for="option in compareSnapshotList" :key="option.value" :value="option.value"
+                        :label="option.label" :title="option.description" class="text-sm"
+                    >
+                        {{ option.label }}
+                    </ff-dropdown-option>
+                </ff-dropdown>
+                <ff-button
+                    v-if="true"
+                    :disabled="!compareSnapshot"
+                    data-action="compare-snapshots"
+                    kind="secondary"
+                    style="height: 30px; width: 106px"
+                    class="w-32"
+                    @click="renderComparison"
+                >
+                    Compare
+                </ff-button>
+            </div>
+            <div v-if="changes.length" class="flex justify-between items-center gap-2 mt-2 ml-2">
+                <div class="whitespace-nowrap">Change {{ changeIndex + 1 }} of {{ changes.length }}:</div>
+                <div class="text-sm text-gray-500 flex-grow truncate overflow-ellipsis">
+                    {{
+                        changes[changeIndex].toString()
+                    }}
+                </div>
+                <ff-button kind="secondary" size="small" class="w-14" @click="gotoPreviousDifference">Prev</ff-button>
+                <ff-button kind="secondary" size="small" class="w-14" @click="gotoNextDifference">Next</ff-button>
+            </div>
+            <div v-else class="mt-2">
+                <div class="text-sm text-gray-500 flex-grow truncate overflow-ellipsis ml-2">No differences found</div>
+            </div>
+            <div ref="compareViewer" data-el="ff-flow-compare-view" class="ff-flow-compare-viewer pt-4" @click.stop.prevent>
+              &nbsp;
+            </div>
+        </template>
+        <template #actions>
+            <div class="flex justify-end">
+                <ff-button data-action="dialog-confirm" @click="confirm()">Close</ff-button>
+            </div>
+        </template>
+    </ff-dialog>
+</template>
+<script>
+
+import FlowRenderer from '@flowfuse/flow-renderer'
+
+import SnapshotsApi from '../../api/snapshots.js'
+
+import Alerts from '../../services/alerts.js'
+
+export default {
+    name: 'FlowViewerDialog',
+    props: {
+        title: {
+            type: String,
+            default: ''
+        }
+    },
+    setup () {
+        return {
+        /**
+         * Shows the compare flows dialog and presents the user with a list of snapshots to compare against
+         * @param {{flows: { flows :[]}}} v1Snapshot - A snapshot object as the base for comparison
+         * @param {[{label: String, value: String}]} snapshotList - A list of snapshots to compare against where label is the snapshot name and value is the snapshot id
+         */
+            showCompareSnapshots (v1Snapshot, snapshotList) {
```

Comment:
Since the whole comparison functionality is broken out into its own dialog, is it necessary to use the long descriptive function name?

When this was part of asset dialog it made sense to use a long descriptive name.  It feels superfluous here!

I would probably rename this to just `show` as per asset viewer dialog.

Will leave it up to you.